test/integration/master/audit_test.go

```diff
@@ -205,21 +251,86 @@ func testAudit(t *testing.T, version string) {
 		t.Fatalf("Unexpected error: %v", err)
 	}
 
-	// perform configmap operations
-	configMapOperations(t, kubeclient)
+	if enableMutatingWebhook {
+		if err := createV1beta1MutationWebhook(kubeclient, url+"/mutation"); err != nil {
+			t.Fatal(err)
+		}
+	}
 
-	// check for corresponding audit logs
-	stream, err := os.Open(logFile.Name())
-	if err != nil {
-		t.Fatalf("Unexpected error: %v", err)
+	var lastMissingReport string
+	if err := wait.Poll(500*time.Millisecond, wait.ForeverTestTimeout, func() (bool, error) {
+		// perform configmap operations
+		configMapOperations(t, kubeclient)
+
+		// check for corresponding audit logs
+		stream, err := os.Open(logFile.Name())
+		if err != nil {
+			return false, fmt.Errorf("Unexpected error: %v", err)
+		}
+		defer stream.Close()
+		missingReport, err := utils.CheckAuditLines(stream, getExpectedEvents(level, enableMutatingWebhook), versions[version])
+		if err != nil {
+			return false, fmt.Errorf("Unexpected error: %v", err)
+		}
+		if len(missingReport.MissingEvents) > 0 {
+			lastMissingReport = missingReport.String()
+			return false, nil
+		}
+		return true, nil
+	}); err != nil {
+		t.Fatalf("failed to get expected events -- missingReport: %s, error: %v", lastMissingReport, err)
 	}
-	defer stream.Close()
-	missingReport, err := utils.CheckAuditLines(stream, expectedEvents, versions[version])
-	if err != nil {
-		t.Fatalf("Unexpected error: %v", err)
+}
+
+func getExpectedEvents(level auditinternal.Level, enableMutatingWebhook bool) []utils.AuditEvent {
+	if !enableMutatingWebhook {
+		return expectedEvents
+	}
+
+	var webhookMutationAnnotations, webhookPatchAnnotations map[string]string
+	var requestObject, responseObject bool
+	if level.GreaterOrEqual(auditinternal.LevelMetadata) {
+		webhookMutationAnnotations = map[string]string{}
+		webhookMutationAnnotations[utils.MutationAuditAnnotationPrefix+"auditmutation.integration.test"] = "true"
+	}
+	if level.GreaterOrEqual(auditinternal.LevelRequest) {
+		webhookPatchAnnotations = map[string]string{}
+		webhookPatchAnnotations[utils.PatchAuditAnnotationPrefix+"auditmutation.integration.test"] = `[{"op":"add","path":"/foo","value":"test"}]`
+		requestObject = true
```

Comment:
some godoc oneliners above each block would help to read this. This is a lot of test value construction.